### ISSUE TITLE
How hard would it be to add xpath support to get and find?

### ISSUE THREAD

**Issue by fringd** [cypress-io/cypress#1274]
### Is this a Feature or Bug?
Feature

### Current behavior:
.get('//div') doesn't work
.find('p/span') also doesn't work
### Desired behavior:
it get's the element using document.evaluate just like get and find currently uses document.querySelector.

or to avoid collisions we add getXpath and findXpath or something.

### Motivation
xpath is fairly standard in integration tests, and I'm working on a shared library of xpath selectors for use with selenium, puppeteer, and ideally, cypress.

I'm happy to help with writing this code if people agree it's a useful feature.

**Comment by jennifer-shehane** [comment]
I don't know the full depth of the complexity of the implementation of this, but this is definitely possible.

- `cy.get()` code is custom to cypress: https://github.com/cypress-io/cypress/blob/issue-895/packages/driver/src/cy/commands/querying.coffee#L93
- `.find()` code just passes is along to jQuerys implementation basically: https://github.com/cypress-io/cypress/blob/issue-895/packages/driver/src/cy/commands/traversals.coffee#L5

**Comment by dizymac** [comment]
+1 for XPath support

**Comment by mgalindo** [comment]
+1

**Comment by CliffDavis** [comment]
+1, this would make migrating from an existing selenium framework significantly easier.

**Comment by brian-mann** [comment]
These things already exist. You can use off the shelf tools to convert xpath to css selectors.

https://www.google.com/search?q=xpath+to+selector+converter&oq=xpath+to+selector+converter&aqs=chrome..69i57j0l4.6500j0j4&sourceid=chrome&ie=UTF-8

There are a few NPM modules you can just install and require in cypress.

**Comment by fringd** [comment]
I'm pretty sure xpath is a super-set of css, so any translator from xpath to css would be incomplete. Part of the reason I want xpath is that it's more powerful.

**Comment by fringd** [comment]
for example, in xpath you can select a node depending upon its children, or its text content.

**Comment by JashonBrown** [comment]
+1

**Comment by JMVL64** [comment]
+1

**Comment by dharshinid** [comment]
+1

**Comment by ailadson** [comment]
+1

**Comment by moxventura** [comment]
+1

**Comment by nusco** [comment]
+1. Even in cases where CSS is enough, having Xpath would make some migrations from other frameworks so much easier.

**Comment by szymach** [comment]
Yeah I've just ran into an issue, where a `select()` did not work due to a non-braking space in the inner HTML of the option. In XPath I could just normalize spaces and it would work just fine, so definitely a +1 from me.

**Comment by fmeyszies** [comment]
+1. Adding a data-cy attribute is a nice way, but we should be allowed to use xpath as well if we don't want to add any code specifically for testing.

**Comment by bryantabaird** [comment]
Yes, searching by text content would be great with xpath, since that cannot be done with CSS selectors (it doesn't allow `contains`, see https://stackoverflow.com/questions/1520429/is-there-a-css-selector-for-elements-containing-certain-text?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa)

**Comment by eduhlana** [comment]
+1

**Comment by v-mwalk** [comment]
Am averse to 'me toos' and +1's but xpath capability it a must.  There are a few scenarios where css will not work; the primary one being pointed out by fringd.

Even if it is the XPath 1.0 engine, it would make Cypress fully usable for any webpage...

**Comment by brian-mann** [comment]
@kamituel in your example above why couldn't you just do this?

```js
// find the cell containing the content
cy.contains('td', '20')

// or you could find a higher up element if you wanted to
cy.contains('tr', '20')
```

The vast majority of use cases I've seen are for XPath related to an element containing text and you can accomplish that with `cy.contains()`.

**Comment by brian-mann** [comment]
I just looked it up and apparently there is already a `document.evaluate` which will take an `xpath`. 

You should already be able to do this with Cypress with a custom command.

https://developer.mozilla.org/en-US/docs/Web/API/Document/evaluate

At the very least, adding this to the query methods of Cypress may not be that difficult afterall.

**Comment by brian-mann** [comment]
@bryantabaird went back through to review comments. Per my comment here: you can use `cy.contains()` rather than xpath if you want to select an element or a parent element by its text content. 

We've found `cy.contains()` can achieve what you normally would do with a much more complex `xpath` selector.

https://github.com/cypress-io/cypress/issues/1274#issuecomment-403538946

**Comment by kamituel** [comment]
@brian-mann To follow a real example, I'm working on a web application that has several large tables (for reporting purposes), that can easily have 20-30 columns. In some test cases, I want to assert on contents of just a few out of those 20-30 columns.

Using: `cy.contains('td', '20')` would work, but:

- it doesn't specify in which column I expect to see the value `"20"`
- it leads to not very readable test code that will grow to be hard to maintain

Let's assume, still following my example, that I want to assert on contents of the first and third column, and I'm not interested in the contents of the 2nd:

```js
cy.contains('td', '20')
cy.contains('td', 'Kate')
```

This code will match:

```html
    <tr>
      <td>Kate</td>
      <td>kate@doe.com</td>
      <td>20</td>
    </tr>
```

but it'll also match:

```
    <tr>
      <td>20</td>
      <td>kate@doe.com</td>
      <td>Kate</td>
    </tr>
```

So I can start adding some more specific selectors, or use `data-cy`:

```js
cy.contains('td:nth-of-type(3)', '20')
cy.contains('td:nth-of-type(1)', 'Kate')
```

But it gets tricky quickly - I'll have to update tests as the table grows, or when I remove a column, etc...

Compare it to (a hypothetical code):

```js
cy.get(cell(3, 'Age')).contains('20');
cy.get(cell(3, 'Username')).contains('Kate');

// With a helper function:

function cell(rowNumber, columnHeader) {
  return `//table
            /tbody
              /tr[position()=${ rowNumber }]
                /td[count(//table/thead/tr/th[text()='${ columnHeader }']/preceding-sibling::th) + 1]`;
}
```

Sure, the helper function uses a fairly complex XPath selector, but it's not that bad once one gets used to XPath, and it's still simpler than a helper function/command would be without XPath. Plus, it leads to simple, readable and maintainable test code.

In general, I prefer CSS over XPath any day, as it leads to shorter selectors whose syntax everyone is familiar with. But when given a choice of a complex helper function that would need to iterate over a number of DOM nodes, versus a declarative XPath selector, I strongly prefer XPath.

Also, yes, there is `document.evaluate` I use often to test XPath selectors (when using them with Selenium). I even added a bookmarklet to Chrome which allows me to easily use it on every website:

```js
javascript:(window.find_all = function (query) {     var nodes = [];     var iterator = document.evaluate(query, document);     try {         var node = iterator.iterateNext();         while (node) {             nodes.push(node);             node = iterator.iterateNext();         }     } catch (e) {         console.error("Document tree modified during iteration", e);         return null;     }          return nodes; })();
```

Usage:

```js
find_all(xpath_selector);
```

I would imagine in Cypress, we could accept XPath selectors prefixed with some character. Like in `cy.get(...)`, which accepts aliases with `@` prefix. Or maybe we could just use `/`?

**Comment by brian-mann** [comment]
What I'm saying is that with Cypress **you already have** native DOM access to your window/document.

Just use the code you wrote just now and put it in a custom command so that it finds you the DOM element and then pass that off to `cy.wrap()` so you can then assume the element and then continue to chain off of it via Cypress.

**Comment by kamituel** [comment]
You mean write a custom command that'll accept an XPath selector and yield a DOM element? Sure, we can do that. And I'm pretty sure I eventually will, as we slowly (but surely) go about migrating our Selenium tests to Cypress ;)

Question is - should Cypress' built-in `.get()` support XPath? (or maybe have support for it in a separate command...)

I'd estimate that maybe 5% of selectors I would write in Selenium are XPath. Apparently, 20+ people who upvoted the OP have somewhat similar experiences. Is it enough to warrant having it built into Cypress itself? Not for me do decide :)

If you, the Cypress' core team, will decide it's not worth the effort, would you accept a PR that adds XPath support to `.get()`, or you rather think it belongs elsewhere (an external library)? Do you have plans to provide a "commons" library with stuff that is helpful to a large number of people, but doesn't necessarily belong to the set of core API's?

**Comment by jennifer-shehane** [comment]
@kamituel We would certainly accept a PR that adds support for XPath! We are not against supporting XPath considering the demand from users here. 

Unfortunately, we are a small team, so it is not within our roadmap at this moment to work on this - so @brian-mann suggestions were to help you all with a workaround to use now. 

We also feature custom commands in our [plugins](https://on.cypress.io/plugins), so if that's an easier option than doing a PR - let us know and we can share it there.

**Comment by HGani1** [comment]
Another +1 for Xpath support in the Cypress .get() method

**Comment by GabrielCarrabbaPereira** [comment]
+1

**Comment by eugene-tkachenko** [comment]
any updates on this? -it's pretty useful in term of work with the pretty complex DOM

**Comment by facebookdevbim** [comment]
Hi @brian-mann

I am having trouble understanding what you are saying.

I have this...

1.
```
function getElementByXpath(path) {
  return document.evaluate(
    path,
    document,
    null,
    XPathResult.FIRST_ORDERED_NODE_TYPE,
    null
  ).singleNodeValue;
}
```

and can do this...

```
getElementByXpath('//input[@type="text" and @name="login"]');  //returns a node

```
What do I need to do in cypress to get the equivalent of 
cy.get(selector).type('blah') but using the above? Is it possible?

**Comment by kamituel** [comment]
@facebookdevbim The idiomatic approach would be to either add a new, custom command, such as `cy.getXpath(...)`, or override built-in `cy.get()` to accept XPath expressions.

I opted for the latter approach myself:

```js
Cypress.Commands.overwrite('get', (originalGet, selector, options) => {
  if (typeof selector === 'string' && selector.startsWith('#xpath#')) {
    var nodes = [];
    var iterator = document.evaluate(query, document);
    try {
       var node = iterator.iterateNext();
       while (node) {
         nodes.push(node);
         node = iterator.iterateNext();
       }
    } catch (e) {
      console.error("Document tree modified during iteration", e);
      return null;
    } 
    return nodes;
  } else {
    return originalGet(selector, options);
  }
})
```

I'm using it like this:

```js
cy.get('#xpath#//div')
```

As you can see, I opted for having a prefix `#xpath#` before every XPath selector. You could do without it and for instance rely on the `//` too, probably.

**Note:** consider the code above pseudocode, as I didn't run it. I translated my code to JS by hand just to illustrate the approach.

**Comment by facebookdevbim** [comment]
Managed to get it working by using cy.document in the custom command. Not sure why I needed it though.

```
"use strict";

// helper functions
let getElementByXpath = function(xPathExpression, container = document) {
  return document.evaluate(
    xPathExpression,
    container,
    null,
    XPathResult.FIRST_ORDERED_NODE_TYPE,
    null
  ).singleNodeValue;
};

let getElementsByXpath = function(
  xPathExpression,
  container = document
) {
  let xPathResult = null;
  let element = null;
  let xPathDOM = [];

  xPathResult = document.evaluate(
    xPathExpression,
    container,
    null,
    XPathResult.ORDERED_NODE_ITERATOR_TYPE,
    null
  );

  try {
    element = xPathResult.iterateNext();
    while (element) {
      xPathDOM.push(element);
      element = xPathResult.iterateNext();
    }
  } catch (e) {
    return null;
  }

  return xPathDOM;
};

let getElementsSnapshotByXpath = function(
  xPathExpression,
  container = document
) {
  let xPathResult = null;
  let xPathDOM = [];

  xPathResult = document.evaluate(
    xPathExpression,
    container,
    null,
    XPathResult.ORDERED_NODE_SNAPSHOT_TYPE,
    null
  );

  for (let i = 0; i < xPathResult.snapshotLength; i++) {
    xPathDOM.push(xPathResult.snapshotItem(i));
  }

  return xPathDOM;
};


// Custom comman overwrite in support/commands.js
Cypress.Commands.overwrite("get", (originalFn, selector, options) => {
  let regExpression = /^(\.?\/\/?[*\w])/gim;
  if (typeof selector === "string" && regExpression.test(selector)) {
    cy.document().then(cyDOM => {
      return getElementByXpath(selector, cyDOM.documentElement);
    });
  } else {
    return originalFn(selector, options);
  }
});
```

**Comment by fringd** [comment]
can someone maybe give me a hint? where is cy.get defined? I'm struggling to track it down.

**Comment by jennifer-shehane** [comment]
@fringd cy.get definition within Cypress repo: https://github.com/cypress-io/cypress/blob/develop/packages/driver/src/cy/commands/querying.coffee#L69

**Comment by jennifer-shehane** [comment]
@bahmutov has implemented custom xpath commands as a Cypress plugin that some of you may find useful here: https://github.com/cypress-io/cypress-xpath

**Comment by edumorata** [comment]
Hello everyone, I am using Cypress-xpath Cypress plugin but I don't know if I can do something like this.

I want to wait for an object to disappear, I am trying to put an object in a variable and put it in a loop to know if it is visible or not and leave the loop when it is not.

If I use `cy.xpath ('xpath')` and the object exists, I can wait until it is null, but when Cypress verifies the xpath (now it is not on the screen) it breaks.

Is there any way to enter the internal cypress check and make a loop inside before the cypress verifies the object and breaks the execution? I mean something like:

```
cy.xpath(objectOverwrited).should(($input) => {

            while ($input != null) {
                
                //Be into loop until the object is null
            }
})
```
Or there is a way to do this function 'waitToVanish' (my personal function name)?

I tried too through `document.evaluate(path, document, null, XPathResult.FIRST_ORDERED_NODE_TYPE, null ).singleNodeValue` but cypress doesn't recognize it while developer console of the browser it does

Thank you very much

**Comment by scorpyto** [comment]
In my opinion .xpath is far more used than .get command. Considering it is just an xpath there should be no changes at all so the support won't be that time consuming? What is bothering me is all the test I have right now -> are they going to stop working like in 2 months or so for example or not?

**Comment by jesusiglesias** [comment]
It would be great if Cypress natively supports Xpath :)

**Comment by HoolioIL** [comment]
Unbelievable, we wanted to use Cypress in our company and also buy Cypress Cloud, in these last weeks we have created a demo for this purpose, but we are using xpath because it is easier for us to find elements (and it is easier to migrate the current cases we have), we also have in most cases of the UI reusable components, only the text is different so xpath is a required and essential function for us. If it will not be maintained is a lot the risk that in future versions we have problems, it is a shame...

**Comment by scorpyto** [comment]
> Unbelievable, we wanted to use Cypress in our company and also buy Cypress Cloud, in these last weeks we have created a demo for this purpose, but we are using xpath because it is easier for us to find elements (and it is easier to migrate the current cases we have), we also have in most cases of the UI reusable components, only the text is different so xpath is a required and essential function for us. If it will not be maintained is a lot the risk that in future versions we have problems, it is a shame...

I completely agree! Nowadays developers use more and more dynamic js objects without id or some main attributes and it is impossible to create reliable test cases without xpaths. I guess a lot of people will stop using Cypress if xpath problem starts to appear.

**Comment by matbgn** [comment]
Incredible, we migrated from TagUI to Cypress for about 3 months, and now we encounter this issue, meaning that we have to migrate all xpath "legacy" call. Sad...

**Comment by hansyohan** [comment]
will there be any future update on this? I mean will Cypress support this natively?

**Comment by atalanchuk-cafemedia** [comment]
Please add the XPath support back

**Comment by Hellsfoul** [comment]
Without xpath, Cypress is not usable for us anymore. In our project, we test a website from a contractor. Good luck to convince them, to add ids everywhere, nevertheless the time it takes until the complex system has them everywhere. CSS is not enough to find the elements needed. Sometimes we just need the //div[contains(text(), "myText")] xpath, to find the correct element.

**Comment by bahmutov** [comment]
You do have cy.contains command …Sent from my iPhoneOn Aug 9, 2023, at 09:32, Hellsfoul ***@***.***> wrote:﻿
Without xpath, Cypress is not usable for us anymore. In our project, we test a website from a contractor. Good luck to convince them, to add ids everywhere, nevertheless the time it takes until the complex system has them everywhere. CSS is not enough to find the elements needed. Sometimes we just need the //div[contains(text(), "myText")] xpath, to find the correct element.

**Comment by mojiAh** [comment]
@bahmutov but cy.contains cannot handle xpath or it can? Isn't it easier to bring back cypress-xpath again?

**Comment by bahmutov** [comment]
No, but you can chain cy.get, cy.contains and find elements by text, no xpath necessary Sent from my iPhoneOn Aug 9, 2023, at 11:37, Moji ***@***.***> wrote:﻿
@bahmutov but cy.contains cannot handle xpath or it can? Isn't it easier to bring back cypress-xpath again?

**Comment by mojiAh** [comment]
@bahmutov It is true when I'm starting a project, but in our case transitioning from xpath in our existing tests means we should rewrite pretty much all the tests which it doesn't make any sense(it does make sense if we get more value but simply we don't).
Can you explain the reason of deprecating the @cypress/xpath while it did make sense to have it in first place?

**Comment by Hellsfoul** [comment]
What about //parent::div  ?
When I remember correctly, I also had issues with chaining too many cypress commands after cy.get
In additon it is bad, when I have to use the locator multiple times in the code. Then, I have to chain the same cypress commands multiple times.

**Comment by scorpyto** [comment]
> contains

First of all  "contains" is so useless when you have multiple similar elements and second -> chain commands are kinda broken, especially with .get command.

**Comment by luciarodriguez** [comment]
+1 from me, even more now that cypress/xpath is no longer supported.

**Comment by Sindhuboston** [comment]
+1 for xpath please

**Comment by satkunas** [comment]
+1 for xpath support

**Comment by betole** [comment]
1+ for xpath support.

**Comment by umeetiusbaar** [comment]
+1 for xpath support

**Comment by jgonzalezalbisu** [comment]
A very large part of the Cypress users will appreciate that this is prioritized. Thanks!

**Comment by dil-ddasmohapatra** [comment]
+1 for xpath support. It's realy helps a lot while integrating with UI elements.

**Comment by mateusjsilva** [comment]
+1 for xpath support

**Comment by kiranHanumantharaya** [comment]
+1 for xpath support.

**Comment by wieben** [comment]
+1
Coming from Selenium to Cypress testing, what seems hard in Cypress is targeting the below button with comment in this piece of DOM
```
<div>
    <div data-w-component="my-panel">
        <div>
            <div><span>Panel title to ignore</span></div>
        </div>
        <div>
            <div>
                <button>Click here</button>
            </div>
        </div>
    </div>
    <div data-w-component="my-panel">
        <div>
            <div><span>Panel title</span></div>
        </div>
        <div>
            <div>
                <button>do not Click here</button>
            </div>
        </div>
        <div>
            <div>
                <button>Click here</button> <!-- this button should be targeted -->
            </div>
        </div>
    </div>
</div>
```
The xpath that would do it
```
//*[@data-w-component="my-panel" and .//*[text()="This panel"]]//*[text()="Click here"]
```
It allows the button to be targeted leaving the DOM to be able to change in all possible ways, except changes to data-w-component attribute, or changes to the title of the panel, or changes to the text of the button.
If Cypress can do that with as little constraint on changes to the DOM, then xpath would not be missed that badly.

**Comment by wieben** [comment]
> If Cypress can do that

It can do something close. For the example in above post the below code clicks the desired button. This Cypress test will break if another panel is nested around the title, whereas the Selenium/XPath test would keep passing.

```
cy.get('[data-w-component="my-panel"]')
            .contains('*', /^This panel$/)
            .parents('[data-w-component="my-panel"]')
            .contains('*', /^Click here$/)
            .click()
```

**Comment by dqiubread** [comment]
how has this not been prioritized since the xpath package has been deprecated?

**Comment by allancostaquality** [comment]
+1 for Cypress to be able to support the use of xpath natively. It seems that the Cy team’s thinking is only geared towards development, where only the simplest and fastest is considered. Xpath is undoubtedly a selector infinitely more powerful than the CSS Selector.

Not to mention teams that would like to migrate from other frameworks and already use xpath and won’t break their heads to change all selectors to CSS.

**Comment by alexsch01** [comment]
As Salesforce Lightning element testing requires xpath (https://github.com/cypress-io/cypress/issues/29294) and the xpath package has been deprecated, I give +1 for this issue
(I would usually not comment just to give +1, but this issue is different)

**Comment by dqiubread** [comment]
I've been having same issues as @alexsch01 here with testing Salesforce Lightning Elements and sometimes the shadow elements are not recognized so we have to use xpath. @bahmutov, @jennifer-shehane as Salesforce is custom LWC made by salesforce which will not have data-testids and use a ton of shadow containers is there any support for this?

**Comment by bahmutov** [comment]
Shadow support is different from xpath. On the personal note regarding xpath selectors: don’t use them. Sent from my iPhoneOn Apr 9, 2024, at 10:14, dqiubread ***@***.***> wrote:﻿
I've been having same issues as @alexsch01 here with testing Salesforce Lightning Elements and sometimes the shadow elements are not recognized so we have to use Cypress. @bahmutov, @jennifer-shehane as Salesforce is custom LWC made by salesforce which will not have data-testids and use a ton of shadow containers is there any support for this?

**Comment by dqiubread** [comment]
@bahmutov, the problem for us is that in our salesforce application its really hard to tell what is and isnt a shadow dom element since salesforce does not explicitly tell you if the element type is inside a shadow element. Instead of for me requesting for a feature on shadow dom elements where idk where the issue is coming from, it would be easier to get xpath support as developer tools already supports it so to me it feels more of a universal solution.

**Comment by bahmutov** [comment]
Can’t you set cypress option to traverse the shadow borders by default?Sent from my iPhoneOn Apr 9, 2024, at 10:41, dqiubread ***@***.***> wrote:﻿
@bahmutov, the problem for us is that in our salesforce application its really hard to tell what is and isnt a shadow dom element since salesforce does not explicitly tell you if the element type is inside a shadow element. Instead of for me requesting for a feature on shadow dom elements where idk where the issue is coming from, it would be easier to get xpath support as developer tools already supports it so to me it feels more of a universal solution.

**Comment by dqiubread** [comment]
@bahmutov, I turned the includeShadowDom to true to try to resolve the issue, but for some reason for some of the elements like these elements where they have an aura class or aura id in salesforce, it sometimes fails to get the element by the classname or id. I was told that the elements are alls built from salesforce so not sure how much custom stuff the development team has done. The reason my team uses xpath is we are able to get these elements that have these aura classes which cy.get and cy.contains for some reason is not able to handle for Saleforce
<img width="469" alt="image" src="https://github.com/cypress-io/cypress/assets/88005136/69a71d5e-bc73-4aa3-a6b3-42aab34350a6">

**Comment by DMaguireKane** [comment]
Seems we can add the now removed @cypress/xpath code to our commands.js to get back the `cy.xpath(...)` functionality, just copy&paste the following into commands.js : [@cypress/xpath index.js](https://github.com/cypress-io/cypress/blob/0af6936c63a066c8fa6ff65ae5e10a5def44a858/npm/xpath/src/index.js)
* doesn't cover .get and .find but better than no xpath support

**Comment by dqiubread** [comment]
> Seems we can add the now removed @cypress/xpath code to our commands.js to get back the `cy.xpath(...)` functionality, just copy&paste the following into commands.js : [@cypress/xpath index.js](https://github.com/cypress-io/cypress/blob/0af6936c63a066c8fa6ff65ae5e10a5def44a858/npm/xpath/src/index.js)
> 
> * doesn't cover .get and .find but better than no xpath support

I'm not totally understanding. Wasn't .get and .find never supported? Also Cypress xpath has been removed from the master branch, do you mean we can just make a custom command using what existed of the xpath functionality from before?

**Comment by DMaguireKane** [comment]
> do you mean we can just make a custom command using what existed of the xpath functionality from before?

Yes, copy&paste the full content of the @cypress/xpath index.js linked above (which is linked to the commit before they removed the content) into your command.js and it appears to restore `cy.xpath(...)` capability (at least in my minimal usage of it so far). It's a crappy workaround for lack of xpath support and potentially will break in the future. I then mentioned that it doesn't cover .get and .find simply as this git issue is specifically about adding xpath support to .get and .find.

**Comment by jgonzalezalbisu** [comment]
> > do you mean we can just make a custom command using what existed of the xpath functionality from before?
> 
> Yes, copy&paste the full content of the @cypress/xpath index.js linked above (which is linked to the commit before they removed the content) into your command.js and it appears to restore `cy.xpath(...)` capability (at least in my minimal usage of it so far). It's a crappy workaround for lack of xpath support and potentially will break in the future. I then mentioned that it doesn't cover .get and .find simply as this git issue is specifically about adding xpath support to .get and .find.

@DMaguireKane despite it has been deprecated, is still available for download in the npmjs, why do you suggest we do this to use `cy.xpath()`command?

**Comment by DMaguireKane** [comment]
@jgonzalezalbisu Ah, I somehow assumed that it was removed because it no longer worked. Apologies for the wasted notifications. Installing `@cypress/xpath` and adding `require('@cypress/xpath')` to command.js still works fine (and add "@cypress/xpath" to tsconfig.json types for intellisense).

**Comment by jgonzalezalbisu** [comment]
> @jgonzalezalbisu Ah, I somehow assumed that it was removed because it no longer worked. Apologies for the wasted notifications. Installing `@cypress/xpath` and adding `require('@cypress/xpath')` to command.js still works fine (and add "@cypress/xpath" to tsconfig.json types for intellisense).

@DMaguireKane nothing to apologize for!
